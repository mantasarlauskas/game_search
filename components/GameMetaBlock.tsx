import React, { ReactNode } from 'react';
import styles from 'styles/game-meta-block.module.scss';

function GameMetaBlock({ title, content }: GameMetaBlockProps) {
    return (
        <div className={styles.root}>
            <div className={styles.title}>
                {title}
            </div>
            {content}
        </div>
    );
}

interface GameMetaBlockProps {
    title: string;
    content: ReactNode;
}

export default GameMetaBlock;
