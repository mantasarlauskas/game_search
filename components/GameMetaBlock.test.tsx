import { render, screen } from '@testing-library/react';
import GameMetaBlock from 'components/GameMetaBlock';

describe('<GameMetaBlock />', () => {
    const props = {
        title: 'title',
        children: <span>children</span>,
    };

    it('matches snapshot', () => {
        const { container } = render(<GameMetaBlock {...props} />);
        expect(container).toMatchSnapshot();
    });

    it('shows title with children', () => {
        render(<GameMetaBlock {...props} />);
        expect(screen.getByText('title')).toBeInTheDocument();
        expect(screen.getByText('children')).toBeInTheDocument();
    });
});
