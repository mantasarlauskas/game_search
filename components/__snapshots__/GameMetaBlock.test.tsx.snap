// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameMetaBlock /> matches snapshot 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="title"
    >
      title
    </div>
    <span>
      children
    </span>
  </div>
</div>
`;
